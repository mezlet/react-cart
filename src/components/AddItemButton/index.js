import React from 'react';
import PropTypes from 'prop-types';
import {Button} from 'react-bootstrap'

const AddItemButton = ({click}) => {
    return (
        <Button className="mt-2 mx-3 custom-green" size="md" onClick={click}>Add to cart</Button>
    )
}

AddItemButton.propTypes = {
    click: PropTypes.func.isRequired
}
export default AddItemButton
