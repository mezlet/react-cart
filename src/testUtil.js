import checkPropTypes from 'check-prop-types';

export const checkProps = (component, expectedProps) => {
    return checkPropTypes(component.propTypes, expectedProps, 'props', component.name)
}