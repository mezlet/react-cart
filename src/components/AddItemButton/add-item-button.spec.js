import React from 'react';
import { shallow } from 'enzyme';
import AddItemButton from './index';
import { checkProps} from '../../testUtil';

const props = {
    click: jest.fn()
}

const setUp = (props = {}) => {
    const component = shallow(<AddItemButton {...props} />);
    return component;
}

describe('AddItemButton', () => {
    let component;
    beforeEach(() => {
        component = setUp(props);
    })
    it('should render without errors', () => {
        expect(component).toMatchSnapshot(); 
    });
    it('should render button', () => {
        const wrapper = component.find(`[data-test='add-item-button']`);
        expect(wrapper.length).toBe(1); 
    })
    it('Checking PropTypes should not throw error', () => {
        const propErr = checkProps(AddItemButton, props);
        expect(propErr).toBeUndefined();
    })
    it('should render click', () => {
        const button = component.find(`[data-test='add-item-button']`);
        button.simulate('click');
        expect(props.click.mock.calls.length).toBe(1);
    })
})