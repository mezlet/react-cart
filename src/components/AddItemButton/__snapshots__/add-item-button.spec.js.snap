// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddItemButton should render without errors 1`] = `ShallowWrapper {}`;
